functional_tests/scylla_operator/test_functional.py

```diff
@@ -503,3 +503,72 @@ def test_deploy_helm_with_default_values(db_cluster: ScyllaPodCluster):
     finally:
         db_cluster.k8s_cluster.helm(f"uninstall {target_chart_name} --timeout 120s", namespace=namespace)
         db_cluster.k8s_cluster.kubectl(f"delete namespace {namespace}")
+
+
+def test_scylla_yaml_override(db_cluster, scylla_yaml):
+    """
+    Test of applying scylla.yaml via configmap
+    - update parameter that exists in scylla.yaml
+    - add parameter
+    """
+
+    with scylla_yaml() as props:
+        configmap_scylla_yaml_content = props
+
+    original_hinted_handoff_throttle_in_kb, new_hinted_handoff_throttle_in_kb = None, None
+
+    with db_cluster.nodes[0].actual_scylla_yaml() as props:
+        original_hinted_handoff = dict(props).get("hinted_handoff_enabled")
+        if not configmap_scylla_yaml_content.get("hinted_handoff_throttle_in_kb"):
+            original_hinted_handoff_throttle_in_kb = dict(props).get("hinted_handoff_throttle_in_kb")
+
+    log.info("configMap's acylla.yaml = %s", configmap_scylla_yaml_content)
+
+    assert isinstance(original_hinted_handoff, bool), (
+        f"configMap scylla.yaml have unexpected 'hinted_handoff_enabled' value: {original_hinted_handoff}")
+    new_hinted_handoff = not original_hinted_handoff
+
+    if original_hinted_handoff_throttle_in_kb:
+        assert isinstance(original_hinted_handoff_throttle_in_kb, int), (
+            f"Node scylla.yaml have unexpected 'hinted_handoff_throttle_in_kb' value: "
+            f"{original_hinted_handoff_throttle_in_kb}")
+        new_hinted_handoff_throttle_in_kb = original_hinted_handoff_throttle_in_kb * 2
+
+    with scylla_yaml() as props:
+        props['hinted_handoff_enabled'] = new_hinted_handoff
+        if new_hinted_handoff_throttle_in_kb:
+            props['hinted_handoff_throttle_in_kb'] = new_hinted_handoff_throttle_in_kb
```

Comment:
Do we really need to update 2 options? Wouldn't just one be enough?

Reply:
It's 2 cases: 
1. update existent param
2. add new one

Reply:
For the moment test doesn't guarantee that it does both operations (add, update).
So, if you want to have exactly both operations then need to make it be applicable always.
Here we have 2 restarts so, we can update there and back any field and we can do "add-remove" as well as "remove-add". So, this test can be improved to guarantee it using the same number of restarts.

Reply:
changed